### PR TITLE
Adde active status to Regions, Cities, Clubs

### PR DESCRIPTION
**Pull request by andrii52**
archiving and unarchiving
Regions, Cities, Clubs

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/pages/Club/Club/CheckActiveMembersForm.less [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    .cityMemberItem :hover {
+        text-decoration: underline;
+    }
+
+    .cityName {
+        text-align: center;
+        align-items: center;
+        -webkit-line-clamp: 1;
+
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.less:16]: delete space

File: src/pages/City/City/Form.module.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    .inputField{
+    width: 100%;
+    }
+  }
+
+  .selectField {
+    float: right;
+    text-align: left;
+    width:320px;
+
```
Comment (christykobyletska) [inline src/pages/City/City/Form.module.css:18]: delete space

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+         htmlType="submit" 
+        >
+         OK
+        </Button>
+
+      </Form.Item> 
+        
+     </Form>;
+
+
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx:113]: delete unnecessary spaces

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import { redirectCities, removeRegion } from "../../../api/regionsApi";
+import { useParams, useHistory, Link, useRouteMatch } from 'react-router-dom';
+import "./CheckActiveMembersForm.less";
+import Title from 'antd/lib/skeleton/Title';
+type CheckActiveCitiesForm = {
+    onAdd: () => void;
+}
+
+
+
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx:13]: delete unnecessary spaces

File: src/pages/City/Cities/ActionCities.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { Switch } from 'antd';
+import React, { useState } from 'react';
+import UserApi from "../../../api/UserApi";
+import { Roles } from "../../../models/Roles/Roles";
+import { useEffect } from 'react';
+import SortedCities from './SortedCities';
+
+
```
Comment (christykobyletska) [inline src/pages/City/Cities/ActionCities.tsx:8]: delete one space

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    const { id } = useParams();
+    const { url } = useRouteMatch()
+    const history = useHistory();
+
+    const [activeCities, setActiveCities] = useState<any[]>([]);
+    const handleSubmit = async ()=>{
+        onAdd();
+        }
+       
+
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx:28]: delete one space

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            ) : null}
+            {props.admins.length !== 0 ? (
+                    <React.Fragment>
+                        <PageHeader 
+                        title = "Спочатку відкріпіть членів проводу: "
+                         className="site-page-header" 
+                        />
+                            <Row className= "cityItems" justify="center">
+                            
+
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx:66]: and here

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+         htmlType="submit" 
+        >
+         OK
+        </Button>
+
+      </Form.Item> 
+        
+     </Form>;
+
+
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx:114]: and here

File: src/pages/Club/Club/Form.module.css [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    .inputField{
+    width: 100%;
+    }
+  }
+
+  .selectField {
+    float: right;
+    text-align: left;
+    width:320px;
+
```
Comment (christykobyletska) [inline src/pages/Club/Club/Form.module.css:18]: delete space

File: src/pages/Regions/CheckActiveCitiesForm.less [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+    .cityMemberItem :hover {
+        text-decoration: underline;
+    }
+
+    .cityName {
+        text-align: center;
+        align-items: center;
+        -webkit-line-clamp: 1;
+
```
Comment (christykobyletska) [inline src/pages/Regions/CheckActiveCitiesForm.less:17]: delete space

File: src/pages/Club/Club/CheckActiveMembersForm.less (hunk 1 above)
Comment (andrii52) [reply src/pages/Club/Club/CheckActiveMembersForm.less]: Done

File: src/pages/City/Cities/ActionCities.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    const [isAdmin, setIsAdmin] = useState<boolean>();
+    
+    const checkAccessToManage = () => {
+        let roles = UserApi.getActiveUserRoles();
+        setIsAdmin(roles.includes(Roles.Admin));
+    }
+
+    const onChange = () => {
+            setSwitchValue(!SwitchValue);
+        }
```
Comment (christykobyletska) [inline src/pages/City/Cities/ActionCities.tsx:23]: closing brace should have the same tabulation as the opening one

File: src/pages/City/Cities/SortedCities.tsx [hunk 12]
```diff
@@ -12,37 +12,49 @@ import Search from "antd/lib/input/Search";
...
+  const [activeCanCreate, setActiveCanCreate] = useState<boolean>(false);
 
   const setPhotos = async (cities: CityProfile[]) => {
-    for await (const city of cities) {
-      if (city.logo === null) {
-        city.logo = CityDefaultLogo;
-      } else {
-        const logo = await getLogo(city.logo);
-        city.logo = logo.data;
+    try{
```
Comment (christykobyletska) [inline src/pages/City/Cities/SortedCities.tsx:39]: put space after try

File: src/pages/City/Cities/SortedCities.tsx [hunk 13]
```diff
@@ -12,37 +12,49 @@ import Search from "antd/lib/input/Search";
...
+      for await (const city of cities) {
+        if (city.logo === null) {
+          city.logo = CityDefaultLogo;
+        } else {
+          const logo = await getLogo(city.logo);
+          city.logo = logo.data;
+        }
       }
+    }finally{
```
Comment (christykobyletska) [inline src/pages/City/Cities/SortedCities.tsx:48]: and here spaces

File: src/pages/City/Cities/SortedCities.tsx [hunk 14]
```diff
@@ -68,17 +100,54 @@ const Cities = () => {
...
+              </Card>
+        </Link>
+          ))   
+    }
+    return null;
+};
+
+  useEffect(() => {
+    switcher ? (getNotActiveCities()):(getActiveCities()) 
+
```
Comment (christykobyletska) [inline src/pages/City/Cities/SortedCities.tsx:136]: need this space?

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+                        ))}
+                        </Row> 
+                </React.Fragment>
+         ) : null}
+            {props.admins.length !== 0 ? (
+                <React.Fragment>
+                <PageHeader 
+             title = "Необхідно відкріпити членів проводу: "
+             className="site-page-header" 
+             />
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: add tabulation for title, classname, and closing tag, so it is on the same tab as opening tag

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+            <Row className= "cityItems" justify="center">
+                {props.admins.map((admin: any) => ( 
+                    <Link to={"/userpage/main/"+admin.userId} target="_blank">
+                        <Col  key={admin.id} >
+                    <div>
+                          <Avatar size={64} src={admin.user.imagePath} />
+                      <p>{admin.user.firstName}</p>
+                    </div>
+                  </Col>
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: and here please add tabulation (for p, col)

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+                    </Link>
+                    ))}
+            </Row> 
+        </React.Fragment>
+        ) : null}
+        {props.followers.length !== 0 ? (
+             <React.Fragment>
+                <PageHeader 
+                title = "Необхідно відкріпити прихильників:"
+                className="site-page-header"
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: please add tabulation for props(title, classname)

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+             <React.Fragment>
+                <PageHeader 
+                title = "Необхідно відкріпити прихильників:"
+                className="site-page-header" 
+                />
+                    <Row className= "cityItems" justify="center">
+                    {props.followers.map((follower: any) => ( 
+                        <Link to={"/userpage/main/"+follower.userId} target="_blank">
+                            <Col
+                            key={follower.id}
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: add here tab

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+                    </Row> 
+                </React.Fragment>
+         ) : null}
+         </Form.Item>
+
+         <Form.Item style = {{ textAlign: "center"}}>
+
+        <Button
+         type="primary" 
+         htmlType="submit"
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: and for these props

File: src/pages/City/City/CheckActiveMembersForm.tsx [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+
+        <Button
+         type="primary" 
+         htmlType="submit" 
+        >
+         OK
+        </Button>
+
+      </Form.Item> 
+
```
Comment (christykobyletska) [inline src/pages/City/City/CheckActiveMembersForm.tsx]: do you need these space after button, form.item closing tags)?

File: src/pages/City/City/CheckActiveMembersForm.tsx (hunk 3 above)
Comment (christykobyletska) [reply src/pages/City/City/CheckActiveMembersForm.tsx:93]: add here these 2 spaces?

File: src/pages/City/City/City.tsx [hunk 21]
```diff
@@ -511,14 +577,34 @@ const City = () => {
...
+                      </Col>) : (
+                        <React.Fragment>
+                          <Col offset={1}>
+                            <Tooltip title="Видалити станицю">
+                              <DeleteOutlined
+                                className="cityInfoIconDelete"
+                                onClick={() => seeDeleteModal()} 
+                              />
+                            </Tooltip>
+                            </Col>
```
Comment (christykobyletska) [inline src/pages/City/City/City.tsx:591]: here shift+tab for /col

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+            onFinish={handleSubmit}
+            form = {form}
+            >
+        <Form.Item
+        className={classes.formField}
+        > {props.members.length !== 0 ? (
+                <React.Fragment>
+                    <PageHeader 
+                    title = "Спочатку відкріпіть членів: "
+                    className="site-page-header"
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: please add tab for props

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+                <React.Fragment>
+                    <PageHeader 
+                    title = "Спочатку відкріпіть членів: "
+                    className="site-page-header" 
+                    />
+                        <Row className= "cityItems" justify="center">
+                            {props.members.map((member: any) => ( 
+                                <Link to={"/userpage/main/"+member.userId} target="_blank">
+                                    <Col
+                                    key={member.id}
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: tab

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+                                </Link>
+                            ))}
+                        </Row> 
+                </React.Fragment>
+            ) : null}
+            {props.admins.length !== 0 ? (
+                    <React.Fragment>
+                        <PageHeader 
+                        title = "Спочатку відкріпіть членів проводу: "
+                         className="site-page-header"
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: tab for props

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+                            
+                                
+                                {props.admins.map((admin: any) => ( 
+                                    <Link to={"/userpage/main/"+admin.userId} target="_blank">
+                                        <Col  key={admin.id} >
+                                    <div>
+                                        <Avatar size={64} src={admin.user.imagePath} />
+                                    <p>{admin.user.firstName}</p>
+                                    </div>
+                                </Col>
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: add tabulation here please

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+                                    
+                            </Row> 
+            </React.Fragment>
+            ) : null}
+            {props.followers.length !== 0 ? (
+                    <React.Fragment>
+                    <PageHeader 
+                title = "Спочатку відкріпіть прихильників: "
+                className="site-page-header" 
+                />
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: add tabulation for props and closing tag

File: src/pages/Club/Club/CheckActiveMembersForm.tsx [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+                    ))}                        
+                </Row> 
+            </React.Fragment>
+            ) : null}
+         </Form.Item>
+         <Form.Item style = {{ textAlign: "right"}}>
+        <Button
+         type="primary" 
+         htmlType="submit" 
+        >
```
Comment (christykobyletska) [inline src/pages/Club/Club/CheckActiveMembersForm.tsx]: tabulation for props

File: src/pages/Club/Club/Club.tsx [hunk 28]
```diff
@@ -498,15 +564,33 @@ const Club = () => {
...
+                      </Col>) : (
+                        <React.Fragment>
+                          <Col offset={1}>
+                            <Tooltip title="Видалити курінь">
+                              <DeleteOutlined
+                                className="clubInfoIconDelete"
+                                onClick={() => seeDeleteModal()}
+                              />
+                            </Tooltip>
+                            </Col>
```
Comment (christykobyletska) [inline src/pages/Club/Club/Club.tsx:584]: shift+tab for /col

File: src/pages/Club/Clubs/ActionClubs.tsx [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+        checkAccessToManage();
+    },[])
+    
+    return (
+        <div>
+            {isAdmin ? ( 
+            <p className={classes.swapper}> Показати не активні округи : <Switch onChange={onChange}/></p> 
+            ): null }
+            <div className={classes.actionsWrapper}>
+            <SortedClubs switcher={SwitchValue} />
```
Comment (christykobyletska) [inline src/pages/Club/Clubs/ActionClubs.tsx:34]: add tab

File: src/pages/Club/Clubs/SortedClubs.tsx [hunk 30]
```diff
@@ -0,0 +1,217 @@
...
+  const [total, setTotal] = useState(0);
+  const [activeTotal, setActiveTotal] = useState(0);
+  const [notActiveTotal, setNotActiveTotal] = useState(0);
+  const [photosLoading, setPhotosLoading] = useState<boolean>(false);
+  const [searchedData, setSearchedData] = useState("");
+  const [activeUserRoles, setActiveUserRoles] = useState<string[]>([]);
+  const [activeCanCreate, setActiveCanCreate] = useState<boolean>(false);
+
+  const setPhotos = async (clubs: ClubProfile[]) => {
+    try{
```
Comment (christykobyletska) [inline src/pages/Club/Clubs/SortedClubs.tsx:39]: space after try

File: src/pages/Club/Clubs/SortedClubs.tsx [hunk 31]
```diff
@@ -0,0 +1,217 @@
...
+      for await (const club of clubs) {
+        if (club.logo === null) {
+          club.logo = CityDefaultLogo;
+        } else {
+          const logo = await getLogo(club.logo);
+          club.logo = logo.data;
+        }
+      }
+    }finally{
```
Comment (christykobyletska) [inline src/pages/Club/Clubs/SortedClubs.tsx:48]: here after and before finaly

File: src/pages/Club/Clubs/SortedClubs.tsx [hunk 32]
```diff
@@ -0,0 +1,217 @@
...
+              hoverable
+              className="cardStyles"
+              cover={
+                  photosLoading ? (
+                  <Skeleton.Avatar shape="square" active />
+                  ) : (
+                      <img src={city.logo || undefined} alt="City" />
+                  )
+              }
+              onClick={() => history.push(`${url}/${city.id}`)}
```
Comment (christykobyletska) [inline src/pages/Club/Clubs/SortedClubs.tsx:124]: add tab for props

File: src/pages/Regions/CheckActiveCitiesForm.tsx [hunk 33]
```diff
@@ -0,0 +1,75 @@
...
+ 
+                        </Col>
+                    </Link>
+                ))}
+            </Row> 
+         </Form.Item>
+        <Form.Item style = {{ textAlign: "center"}}>
+            <Button
+            type="primary" 
+            htmlType="submit"
```
Comment (christykobyletska) [inline src/pages/Regions/CheckActiveCitiesForm.tsx]: add tabulation for props

File: src/pages/Regions/Region.tsx [hunk 34]
```diff
@@ -155,16 +160,64 @@ const Region = () => {
     setRegionLogoLoading(false);
   };
 
+  const ArchiveRegion = async () => {
+    try {
+      await archiveRegion(region.id);
+    }finally{
```
Comment (christykobyletska) [inline src/pages/Regions/Region.tsx:166]: spaces

File: src/pages/Regions/Region.tsx [hunk 35]
```diff
@@ -450,6 +509,11 @@ const Region = () => {
                   second={url.replace(`/${id}`, "")}
                   second_name="Округи"
                 />
+                {isActiveRegion ? null : (
+                  <Tag className="status" color = {"red"}>
+                  Заархівовано
```
Comment (christykobyletska) [inline src/pages/Regions/Region.tsx:512]: please add tab

File: src/pages/Regions/Region.tsx [hunk 36]
```diff
@@ -596,17 +660,36 @@ const Region = () => {
...
+                            </Col>) : (
+                              <React.Fragment>
+                                <Col offset={1}>
+                                  <Tooltip title="Видалити станицю">
+                                    <DeleteOutlined
+                                      className="cityInfoIconDelete"
+                                      onClick={() => seeDeleteModal()} 
+                                    />
+                                  </Tooltip>
+                                  </Col>
```
Comment (christykobyletska) [inline src/pages/Regions/Region.tsx:679]: shift+tab for /col

File: src/pages/Regions/SortedRegions.tsx [hunk 37]
```diff
@@ -72,19 +93,54 @@ const Regions = () => {
...
     setPage(1);
     setSearchedData(event);
   };
 
+  const renderRegion = (arr: RegionProfile[]) => {
+    if (arr) {
+        return  arr.map((region: RegionProfile) =>(
+          <Link to={`${url}/${region.id}`}>
+              <Card
+              key={region.id}
```
Comment (christykobyletska) [inline src/pages/Regions/SortedRegions.tsx:107]: please tabulate inside <card> </card>

**Review by christykobyletska — CHANGES_REQUESTED** [review]

**Review by christykobyletska — APPROVED** [review]

**Review by OleksandrBilan — APPROVED** [review]

File: src/api/clubsApi.tsx [hunk 38]
```diff
@@ -92,6 +92,34 @@ export const getClubByPage = async (page: number, pageSize: number, clubName: st
...
+export const getActiveClubByPage = async (page: number, pageSize: number, clubName: string | null = null) => {
+  return api
+    .get(`Club/Profiles/Active/${page}`, { page, pageSize, clubName })
+    .catch((error) => {
+      throw new Error(error);
+    });
+};
+
+export const getNotActiveClubByPage = async (page: number, pageSize: number, clubName: string | null = null) => {
+  return api
```
Comment (voroniak) [inline src/api/clubsApi.tsx:104]: Add await

File: src/api/clubsApi.tsx [hunk 39]
```diff
@@ -92,6 +92,34 @@ export const getClubByPage = async (page: number, pageSize: number, clubName: st
...
+    .get(`Club/Profiles/NotActive/${page}`, { page, pageSize, clubName })
+    .catch((error) => {
+      throw new Error(error);
+    });
+};
+
+export const archiveClub = async (id: number) => {
+  return api.put(`Club/ArchiveClub/${id}`).catch((error) => {
```
Comment (voroniak) [inline src/api/clubsApi.tsx:112]: The same

**Review by voroniak — CHANGES_REQUESTED** [review]

File: src/api/clubsApi.tsx (hunk 38 above)
Comment (andrii52) [reply src/api/clubsApi.tsx]: Done

**Review by nicoolle — APPROVED** [review]

**Review by voroniak — APPROVED** [review]

**Pull request merged**